workspaces/mi/mi-extension/src/rpc-managers/mi-diagram/rpc-manager.ts

```
@@ -6074,6 +6082,68 @@ ${keyValuesXML}`;
+
+                if (!response || !response.columns || !response.columns.length) {
+                    resolve({ columns: [] });
+                    return;
+                }
+
+                resolve(response);
+            } catch (error) {
+                console.error(`Error getting dynamic fields: ${error}`);
+                resolve({ columns: [] });
+            }
+        });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)